modules/transport-netty4/src/main/java/org/opensearch/http/netty4/Netty4HttpServerTransport.java

```diff
@@ -531,4 +532,21 @@ public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) {
             }
         }
     }
+
+    /**
+     * Extension point that allows a NetworkPlugin to extend the netty pipeline and inspect headers after request decoding
+     */
+    protected ChannelInboundHandlerAdapter createHeaderVerifier() {
```

Comment:
Does NIO have the equivalent of the netty pipeline? This is specific to a step in the netty pipeline. 

How could this look like being abstracted to `AbstractHttpServerTransport`?

Reply:
So the implementation would be surely up to the engine, but
```
interface HttpHeaderVerifier {
  ...
}

class AbstractHttpServerTransport {
    protected Optional<HttpHeaderVerifier> headerVerifier() {
        ....
    }
}

```

So Netty would wrap `HttpHeaderVerifier` into channel, but other engines will use different means

Reply:
I'm not sure it will correctly apply to other transport systems.  This change is very directly tied to how decompression is handled (or not) in the netty ecosystem.  While it might make sense to create an abstraction that is reusable for `nio` mechanism, I'm not sure that is needed at this point.

I'd rather start smaller with this change and then deprecate this change if there is a better way with an abstraction.

Reply:
We discussed the matter briefly with @cwperks , the implementation how verifier kicks is engine specific, but the logic it does by verifying headers is not.  I would need that for https://github.com/opensearch-project/OpenSearch/pull/9672 as well, not only `nio`